o1vm/src/pickles/main.rs

```diff
@@ -65,11 +69,31 @@ pub fn main() -> ExitCode {
     // Initialize the environments
     let mut mips_wit_env =
         mips_witness::Env::<Fp, PreImageOracle>::create(cannon::PAGE_SIZE as usize, state, po);
-    let mut _mips_con_env = mips_constraints::Env::<Fp>::default();
+
+    // TODO: give this to the prover + verifier
+    let _constraints = {
+        let mut mips_con_env = mips_constraints::Env::<Fp>::default();
+        let mut constraints = Instruction::iter()
+            .flat_map(|instr_typ| instr_typ.into_iter())
+            .fold(vec![], |mut acc, instr| {
+                interpreter::interpret_instruction(&mut mips_con_env, instr);
+                let selector = mips_con_env.get_selector();
+                let constraints_with_selector: Vec<E<Fp>> = mips_con_env
+                    .get_constraints()
+                    .into_iter()
+                    .map(|c| selector.clone() * c)
+                    .collect();
```

Comment:
I don't think this is correct, since this would iterate over each instruction once.  We want to iterate over the instructions that were actually run in the MIPS program, not every instruction once, in order.

Reply:
I understand now: this is just for the constraints.